api/samples/inscount.cpp

```diff
@@ -164,9 +164,38 @@ event_bb_analysis(void *drcontext, void *tag, instrlist_t *bb, bool for_trace,
             }
         }
     }
-    /* Count instructions */
+    /* Count instructions. If this client is being run with a client which
+     * emulates instructions, then both native and emulated instructions will be
+     * counted. Each emulated intruction is replaced by a series of native
```

Comment:
> We want it to only count the original native, right? Not the emulated replacement.

I think we should count each emulated instruction (but not its native replacements). To be clear, I think we should count native instructions, emulated instructions but NOT the native instructions which are used to emulate the emulated instructions, otherwise it isn't an accurate count AIUI. I've tried to clarify the comment:

    /* Count instructions. If an emulation client is running with this client,   
     * we want to count all the native instructions and the emulated instruction 
     * but NOT the native instructions used for emulation.                       
     */